<launch>
                                <!-- RVIZ -->

  <!-- Load the URDF to the parameter server -->
  <param name="robot_description" textfile="$(find ver5_bokhop)/urdf/ver5_bokhop.urdf"/>

  <!-- Show in Rviz  -->
  <arg name="use_rviz" default="true" />
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find my_hw_interface)/basic.rviz"/> -->
  <include file="$(find movit_robot_arm_sim)/launch/moveit_rviz.launch" if="$(arg use_rviz)">
    <arg name="rviz_config" value="$(find my_hw_interface)/basic.rviz"/>
    <!-- <arg name="debug" value="$(arg debug)"/> -->
  </include>


                                <!-- ROS CONTROL -->

  <!-- Load controller settings -->
  <rosparam file="$(find movit_robot_arm_sim)/config/ros_controllers.yaml" command="load"/>
  
  <!-- Load hardware interface -->
  <node name="robot_hwi_node" pkg="my_hw_interface" type="robot_hwi_node" output="screen"/>

  <!-- Load controller manager  -->
  <node name= "ros_controller_spawner" pkg= "controller_manager" type="controller_manager" respawn="false" 
  output="screen" args="spawn /ver5_bokhop/joint_state_controller /ver5_bokhop/upper_arm_controller "/>

  <!-- Convert joint states to /tf tranforms -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
    <!-- <remap from="/joint_states" to="/ver5_bokhop/joint_states" /> -->
  </node>



                                 <!-- MOVEIT -->  

   <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <include file="$(find movit_robot_arm_sim)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>

  <!-- Remap follow_joint_trajectory -->
  <remap from="/joint_trajectory_action" to="ver5_bokhop/upper_arm_controller/follow_joint_trajectory"/>

  <!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
  <include file="$(find movit_robot_arm_sim)/launch/move_group.launch">
    <arg name ="allow_trajectory_execution" value="true"/>
    <arg name="publish_monitored_planning_scene" value="true" />
    <!-- <arg name="load_robot_description" value="true"/> -->
    <arg name ="info" value="true"/>
  </include>
  
</launch>